ssz-rs/src/list.rs

```diff
@@ -290,6 +291,23 @@ where
 
 impl<T, const N: usize> SimpleSerialize for List<T, N> where T: SimpleSerialize {}
 
+impl<T, const N: usize> SszReflect for List<T, N>
+where
+    T: SimpleSerialize + SszReflect + AsAny,
+{
+    fn ssz_type_class(&self) -> SszTypeClass {
+        SszTypeClass::Elements(ElementsType::List)
+    }
+
+    fn list_elem_type(&self) -> Option<&dyn SszReflect> {
+        Some(self.index(0))
```

Comment:
so this is going to be a problem for sure, as this won't work for empty lists

Reply:
return `None` if empty? I don't follow exactly what is going on here yet